app/components/ShopPage/Sidebar.tsx

```diff
@@ -1,41 +1,60 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import styled from 'styled-components';
 import theme from '@/styles/theme';
 import SidebarItem from './SidebarItem';
+import { ShopProductType } from 'app/(route)/shop/shopProduct';
+
+interface SidebarProps {
+  selectedOptions: string[];
+  setSelectedOptions: React.Dispatch<React.SetStateAction<string[]>>;
```

Comment:
setState를 props로 바로 넘겨오는 것보단 onSetState으로 함수화해서 넘겨오는 것을 추천드립니다.

setState는 value를 저장하고 수정하는 기능만하고, 그 외의 추가적인 기능은 없는 걸로 알고 있습니다. 그래서 해당 state의 setState 외의 추가적인 로직을 포함하고 싶을 때는 핸들러 함수를 사용하는 것이 좋습니다. (가독성 측면으로도 좋습니다)

```suggestion
  onSetSelectedOptions: (value: string[]) => void;
```